src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -605,6 +731,107 @@ public void withLock(boolean waitForRecovery, ChannelConsumer consumer) throws I
             }
         }
 
+        public void retryingPublishWithLock(ChannelConsumer consumer, ConnectionManagerConfiguration configuration) throws InterruptedException {
+            lock.lock();
+            Iterator<RetryingDelay> iterator = configuration.createRetryingDelaySequence().iterator();
+            try {
+                Channel tempChannel = getChannel(true);
+                while (true) {
+                    try {
+                        consumer.consume(tempChannel);
+                        break;
+                    } catch (IOException | ShutdownSignalException e) {
+                        var currentValue = iterator.next();
+                        int recoveryDelay = currentValue.getDelay();
+                        LOGGER.warn("Retrying publishing #{}, waiting for {}ms. Reason: {}", currentValue.getTryNumber(), recoveryDelay, e);
+                        TimeUnit.MILLISECONDS.sleep(recoveryDelay);
+
+                        // We should not recover the channel if its connection is closed
+                        // If we do that the channel will be also auto recovered by RabbitMQ client
+                        // during connection recovery, and we will get two new channels instead of one closed.
+                        if (!tempChannel.isOpen() && tempChannel.getConnection().isOpen()) {
+                            tempChannel = recreateChannel();
+                        }
+                    }
+                }
+            } finally {
+                lock.unlock();
+            }
+        }
+
+        public <T> T retryingConsumeWithLock(ChannelMapper<T> mapper, ConnectionManagerConfiguration configuration) throws InterruptedException, IOException {
+            Iterator<RetryingDelay> iterator = null;
+            IOException exception;
+            Channel tempChannel = null;
+            boolean isChannelClosed = false;
+            while (true) {
+                lock.lock();
+                try {
+                    if (tempChannel == null) {
+                        tempChannel = getChannel();
+                    } else if (isChannelClosed) {
+                        tempChannel = recreateChannel();
+                    }
+
+                    var tag = mapper.map(tempChannel);
+                    isSubscribed = true;
+                    return tag;
+                } catch (IOException e) {
+                    var reason = tempChannel.getCloseReason();
+                    isChannelClosed = reason != null;
+                    if (isChannelClosed && reason.getMessage().contains("reply-text=RESOURCE_LOCKED")) {
```

Comment:
Why we shouldn't retry in this case?
Our API inconsistent
retryingPublishWithLock(...) throws InterruptedException
retryingConsumeWithLock(...) throws InterruptedException, IOException

Reply:
@Nikita-Smirnov-Exactpro We should not retry in this case because we never will be able to connect to the queue if we receive this error. This error happens if we try to subscribe to an exclusive queue that was created by another connection.
In the previous discussions in this PR we agreed that we should report this and do no try to recover

Reply:
Please add comment for this case